pkg/protocol/sybilprotection/seatmanager/topstakers/topstakers.go

```diff
@@ -0,0 +1,260 @@
+	s.committeeMutex.RLock()
+	defer s.committeeMutex.RUnlock()
+
+	type poolWithAccountID struct {
+		accountID iotago.AccountID
+		pool      *account.Pool
+	}
+
+	candidatePools := make([]*poolWithAccountID, 0)
+	candidates.ForEach(func(id iotago.AccountID, pool *account.Pool) bool {
+		candidatePools = append(candidatePools, &poolWithAccountID{pool: pool, accountID: id})
+
+		return true
+	})
+
+	sort.Slice(candidatePools, func(i, j int) bool {
+		if candidatePools[i].pool.PoolStake != candidatePools[j].pool.PoolStake {
+			return candidatePools[i].pool.PoolStake > candidatePools[j].pool.PoolStake
+		}
+
+		if candidatePools[i].pool.ValidatorStake != candidatePools[j].pool.ValidatorStake {
+			return candidatePools[i].pool.ValidatorStake > candidatePools[j].pool.ValidatorStake
+		}
+
+		// TODO: add more tie-breaking
+
+		// two candidates never have the same account ID because they come in a map
+		return bytes.Compare(candidatePools[i].accountID[:], candidatePools[j].accountID[:]) > 0
+	})
+
+	// Create new Accounts instance that only included validators selected to be part of the committee.
+	accounts := account.NewAccounts()
+	for _, candidatePool := range candidatePools[:s.optsSeatCount] {
+		accounts.Set(candidatePool.accountID, candidatePool.pool)
+	}
+	committee := accounts.SelectCommittee(accounts.IDs()...)
+
+	err := s.committeeStore.Store(epoch, committee.Accounts())
+	if err != nil {
+		return nil, ierrors.Wrapf(err, "error while storing committee for epoch %d", epoch)
+	}
+
+	return committee, nil
+}
+
+// CommitteeInSlot returns the set of validators selected to be part of the committee in the given slot.
+func (s *SeatManager) CommitteeInSlot(slot iotago.SlotIndex) (*account.SeatedAccounts, bool) {
+	s.committeeMutex.RLock()
+	defer s.committeeMutex.RUnlock()
+
+	return s.committeeInEpoch(s.apiProvider.APIForSlot(slot).TimeProvider().EpochFromSlot(slot))
+}
+
+// CommitteeInEpoch returns the set of validators selected to be part of the committee in the given epoch.
+func (s *SeatManager) CommitteeInEpoch(epoch iotago.EpochIndex) (*account.SeatedAccounts, bool) {
+	s.committeeMutex.RLock()
+	defer s.committeeMutex.RUnlock()
+
+	return s.committeeInEpoch(epoch)
+}
+
+func (s *SeatManager) committeeInEpoch(epoch iotago.EpochIndex) (*account.SeatedAccounts, bool) {
+	c, err := s.committeeStore.Load(epoch)
+	if err != nil {
+		panic(ierrors.Wrapf(err, "failed to load committee for epoch %d", epoch))
+	}
+
+	if c == nil {
+		return nil, false
+	}
+
+	return c.SelectCommittee(c.IDs()...), true
+}
+
+// OnlineCommittee returns the set of validators selected to be part of the committee that has been seen recently.
+func (s *SeatManager) OnlineCommittee() ds.Set[account.SeatIndex] {
+	s.activityMutex.RLock()
+	defer s.activityMutex.RUnlock()
+
+	return s.onlineCommittee
+}
+
+func (s *SeatManager) SeatCount() int {
+	s.committeeMutex.RLock()
+	defer s.committeeMutex.RUnlock()
+
+	return int(s.optsSeatCount)
```

Comment:
same as before, what happens if we have less candidates?

Reply:
This should always return the same number as this is the number of validators we will have at all times.